src/vcpkg/base/system.process.cpp

```diff
@@ -693,35 +694,52 @@ namespace vcpkg
         saAttr.lpSecurityDescriptor = NULL;
 
         // Create a pipe for the child process's STDOUT.
-        if (!CreatePipe(&ret.child_stdout, &startup_info.hStdOutput, &saAttr, 0))
+        if (!CreatePipe(&ret.child_stdout, &startup_info_ex.StartupInfo.hStdOutput, &saAttr, 0))
         {
             return format_system_error_message("CreatePipe stdout", GetLastError());
         }
 
-        // Ensure the read handle to the pipe for STDOUT is not inherited.
-        if (!SetHandleInformation(ret.child_stdout, HANDLE_FLAG_INHERIT, 0))
-        {
-            return format_system_error_message("SetHandleInformation stdout", GetLastError());
-        }
-
         // Create a pipe for the child process's STDIN.
-        if (!CreatePipe(&startup_info.hStdInput, &ret.child_stdin, &saAttr, 0))
+        if (!CreatePipe(&startup_info_ex.StartupInfo.hStdInput, &ret.child_stdin, &saAttr, 0))
         {
             return format_system_error_message("CreatePipe stdin", GetLastError());
         }
 
-        // Ensure the write handle to the pipe for STDIN is not inherited.
-        if (!SetHandleInformation(ret.child_stdin, HANDLE_FLAG_INHERIT, 0))
+        startup_info_ex.StartupInfo.hStdError = startup_info_ex.StartupInfo.hStdOutput;
+
+        // Ensure that only the write handle to STDOUT and the read handle to STDIN are inherited.
+        // from https://devblogs.microsoft.com/oldnewthing/20111216-00/?p=8873
+        SIZE_T size = 0;
+        if (InitializeProcThreadAttributeList(nullptr, 1, 0, &size) || GetLastError() != ERROR_INSUFFICIENT_BUFFER)
```

Comment:
I would really like to see an RAII type for this.

Reply:
Only for the clean up or for everything? So should a failure in the constructor (calls InitializeProcThreadAttributeList) throw an exception?

Reply:
I wish it were for the pipes too but that's preexisting so won't ask you to go there. I meant something like:

```
struct ProcAttributeList {
    ExpectedL<ProcAttributeList> create(DWORD dwAttributeCount);
    ExpectedL<Unit> update_attribute(DWORD_PTR Attribute, PVOID lpValue, SIZE_T cbSize);
    ~ProcAttributeList();

    ProcAttributeList(const ProcAttributeList&) = delete;
    ProcAttributeList& operator=(const ProcAttributeList&) = delete;
private:
    ProcAttributeList(/* *** */);
    std::vector<unsigned char> buffer;
};
```

Reply:
Have done this in https://github.com/microsoft/vcpkg-tool/pull/887/commits/a3bbb3090d127f896cc0f2b03acac12b7c2b9e50